### DISCUSSION TITLE
How to use a specific property in template in dataview code block?

### DISCUSSION THREAD

**Discussion by francisHugo** [SilentVoid13/Templater#1417] [Help] [answered]
I had a template file as below, I would like to have a dataview code block to query the note based on the third tag which is selected when the template is applied. I am not sure whether the question should be here or in Dataview repo, since it is a dataview template question.

Question is:

**How to put the third tag in dataview block?**

`````
---
tags: 
  - tag1
  - tag2
  - #<% await tp.system.suggester(["1 tag3", "2 tag4", "3 tag5"], ["smallCarton", "bigCarton", "smallPlasticBox", "bigPlasticBox"], false,"input tags") %>  
createdDate: <% tp.date.now() %>
---


## Dataview Query

```dataview
table createdDate
from the third tag which input when the template is applied
```


`````

The template is planning to generate a note like below, when the template is applied and smallCarton is selected.

````
---
tags: 
  - tag1
  - tag2
  - smallCarton  
createdDate:  2024-07-20
---


## Dataview Query #smallCarton

```dataview
table createdDate
from #smallCarton
```



````

Will anybody give any hints or help on how to handle it? 

TIA

Best Regards

**Comment by francisHugo** [comment]
Also tried in template:

````
```dataview
table createdDate
from #<% await tp.file.tags[2] %>
```
````
and get no results, and when switch to source mode, it says:

````
```dataview
table createdDate
from #undefined
```
````

It keeps with `#undefined` , whether `0` or `1` or `2` is inside the `[ ]` of the `<% await tp.file.tags[ ] %>`

**Comment by Zachatoo** [comment, marked as answer]
The tags for the file won't be available for Templater to retrieve from the metadata cache until after the file is created, which is too late for your template code.

I would recommend storing the data in a variable and referencing it in multiple places in your template, something like this.

````js
<%*
const inputTags = await tp.system.suggester(["1 tag3", "2 tag4", "3 tag5"], ["smallCarton", "bigCarton", "smallPlasticBox", "bigPlasticBox"], false,"input tags");
-%>
---
tags: 
  - tag1
  - tag2
  - #<% inputTags %>  
createdDate: <% tp.date.now() %>
---


## Dataview Query

```dataview
table createdDate
from #<% inputTags %>
```
````

**Reply by francisHugo** [reply]
Yes, this works! thank you very much!
